### PR TITLE
Checkbox Component.

### PR DESCRIPTION
**Pull request by mahima-vs**

### CODE REVIEW

File: examples/ios/examples.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -16,6 +16,25 @@
 		2D02E4BD1E0B4A84006451C7 /* Images.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 13B07FB51A68108700A75B9A /* Images.xcassets */; };
 		2D02E4BF1E0B4AB3006451C7 /* main.m in Sources */ = {isa = PBXBuildFile; fileRef = 13B07FB71A68108700A75B9A /* main.m */; };
 		2DCD954D1E0B4F2C00145EB5 /* examplesTests.m in Sources */ = {isa = PBXBuildFile; fileRef = 00E356F21AD99517003FC87E /* examplesTests.m */; };
+		7E03F422EC8D4DE9B0AD9220 /* AntDesign.ttf in Resources */ = {isa = PBXBuildFile; fileRef = 1842DB8DA29647C3BBAE8FA4 /* AntDesign.ttf */; };
+		AD8E12FA20664B83A867F806 /* Entypo.ttf in Resources */ = {isa = PBXBuildFile; fileRef = A5F5CDDBDFE74A629DB8AFF0 /* Entypo.ttf */; };
+		AB2CCF9B459548AB997D7DCC /* EvilIcons.ttf in Resources */ = {isa = PBXBuildFile; fileRef = 45A17D972E4145D8AB7000A0 /* EvilIcons.ttf */; };
```
Comment (mahima-vs) [inline examples/ios/examples.xcodeproj/project.pbxproj:21]: Please do not commit these files, or put them in .gitignore file.

File: examples/src/theme.js [hunk 2]
```diff
@@ -1,55 +1,152 @@
...
+      fontFamily: style.fontFamily
+        ? style.fontFamily
+        : textInputStyles.style.fontFamily,
+      color: style.color ? style.color : textInputStyles.style.color,
+    },
+  };
+  return customizeTextInputStyle;
+};
+
+// function to change oll style without theme to stylw with theme
```
Comment (mahima-vs) [inline examples/src/theme.js:94]: Improve comments.

File: examples/src/theme.js [hunk 3]
```diff
@@ -1,55 +1,152 @@
...
+      ? style.borderBottomWidth
+      : oldStyle.borderBottomColor,
+    fontFamily: style.fontFamily ? style.fontFamily : oldStyle.fontFamily,
+  };
+  return oldStyle;
+};
 
-    const style = getThemeStyle(theme);
-    const customizeTextInputStyle = {
+export const applyThemeOnCheckboxStyle = (styleWithouthTheme, theme) => {
```
Comment (mahima-vs) [inline examples/src/theme.js:139]: styleWithouthTheme typo.

File: examples/App.js [hunk 4]
```diff
@@ -7,52 +7,35 @@
...
 
 // TODO: will use Uibuilder from npm package later.
-// import UiBuilder from 'react-native-ui-builder';
+//import UiBuilder from 'react-native-ui-builder';
 import React from 'react';
 import UiBuilder from './src/index';
 import {Colors} from 'react-native/Libraries/NewAppScreen';
 import {View, StyleSheet, Text} from 'react-native';
-
+import {CheckBox, Icon} from 'react-native-elements';
```
Comment (mahima-vs) [inline examples/App.js:15]: Remove unused import statements.

File: examples/ios/examples.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -16,6 +16,25 @@
 		2D02E4BD1E0B4A84006451C7 /* Images.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 13B07FB51A68108700A75B9A /* Images.xcassets */; };
 		2D02E4BF1E0B4AB3006451C7 /* main.m in Sources */ = {isa = PBXBuildFile; fileRef = 13B07FB71A68108700A75B9A /* main.m */; };
 		2DCD954D1E0B4F2C00145EB5 /* examplesTests.m in Sources */ = {isa = PBXBuildFile; fileRef = 00E356F21AD99517003FC87E /* examplesTests.m */; };
+		7E03F422EC8D4DE9B0AD9220 /* AntDesign.ttf in Resources */ = {isa = PBXBuildFile; fileRef = 1842DB8DA29647C3BBAE8FA4 /* AntDesign.ttf */; };
```
Comment (mahima-vs) [inline examples/ios/examples.xcodeproj/project.pbxproj:19]: No need to commit "project.pbxproj" files

File: examples/App.js [hunk 6]
```diff
@@ -7,66 +7,43 @@
...
 
 // TODO: will use Uibuilder from npm package later.
-// import UiBuilder from 'react-native-ui-builder';
-
+//import UiBuilder from 'react-native-ui-builder';
 import React from 'react';
 import UiBuilder from './src/index';
-import { View } from 'react-native';
-
+import {Colors} from 'react-native/Libraries/NewAppScreen';
```
Comment (mahima-vs) [inline examples/App.js:13]: Please remove unused imports.

File: examples/src/components/button.js [hunk 7]
```diff
@@ -1,89 +1,96 @@
-/**
- * Sample React Native App
- * https://github.com/facebook/react-native
- *
- * @format
- * @flow strict-local
- */
+// /**
+//  * Sample React Native App
```
Comment (mahima-vs) [inline examples/src/components/button.js:2]: Why are you commented button component.

File: examples/src/theme/checkboxTheme.js [hunk 8]
```diff
@@ -0,0 +1,60 @@
+import {getThemeStyle} from '../utility/utils';
+
+// function to change old style without theme to stylw with theme
```
Comment (mahima-vs) [inline examples/src/theme/checkboxTheme.js:3]: typo, pls check spellings of your comment.

File: examples/src/theme/checkboxTheme.js [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    fontFamily: style.fontFamily ? style.fontFamily : oldStyle.fontFamily,
+  };
+  return oldStyle;
+};
+
+export const applyThemeOnCheckboxStyle = (styleWithouthTheme, theme) => {
+  const newProps = {...styleWithouthTheme};
+  const style = getThemeStyle(theme);
+  // get changes according to theme in three objects
+  let containerStyle = changeStyleContainer(newProps.containerStyle, style);
```
Comment (mahima-vs) [inline examples/src/theme/checkboxTheme.js:52]: We already have discussed this, this piece is not accepted, pls follow the same we were doing for other components.

Comment (saksham695) [reply examples/src/theme/checkboxTheme.js:52]: I have already changed these files , will update soon.

File: examples/src/components/Checkbox.js [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      checked: false,
+      color: '',
+    };
+  }
+
+  componentDidMount() {
+    let colorCopy;
```
Comment (mahima-vs) [inline examples/src/components/Checkbox.js:19]: Use meaningful names of a variable.

File: examples/src/components/Checkbox.js [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+  }
+  handleChange = () => {
+    const initialValue = this.state.checked;
+    const checkedColor = this.props.titleStyle.checkedColor
+      ? this.props.titleStyle.checkedColor
+      : this.props.titleStyle.color;
+    const uncheckedColor = this.props.titleStyle.uncheckedColor
+      ? this.props.titleStyle.uncheckedColor
+      : this.props.titleStyle.color;
+    // console.log('handleChnage', this.props.titleStyle);
```
Comment (mahima-vs) [inline examples/src/components/Checkbox.js:43]: Code cleanup

File: examples/ios/examples.xcodeproj/project.pbxproj (hunk 1 above)
Comment (saksham695) [reply examples/ios/examples.xcodeproj/project.pbxproj:21]: done

File: examples/src/theme.js (hunk 2 above)
Comment (saksham695) [reply examples/src/theme.js]: done

File: examples/src/theme.js (hunk 3 above)
Comment (saksham695) [reply examples/src/theme.js]: done

File: examples/src/components/Checkbox.js [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+        ? this.props.titleStyle.uncheckedColor
+        : this.props.titleStyle.color;
+    }
+    // to set the initial state of checkbox component given by user and color of text content
+    this.setState({
+      checked: this.props.checked,
+      color: colorCopy,
+    });
+  }
+  handleChange = () => {
```
Comment (mahima-vs) [inline examples/src/components/Checkbox.js:35]: wht is the meaning of "handleChange" ? Use meaningful name of methods as well.

File: examples/App.js (hunk 4 above)
Comment (saksham695) [reply examples/App.js:15]: done

File: examples/src/theme/checkboxTheme.js (hunk 9 above)
Comment (saksham695) [reply examples/src/theme/checkboxTheme.js]: done

File: examples/src/components/Checkbox.js [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+    const initialValue = this.state.checked;
```
Comment (mahima-vs) [inline examples/src/components/Checkbox.js:36]: Why we are storing the value in a seperate variable? 
Use this.state.checked wherever reqd.

File: examples/src/components/Checkbox.js [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+    const checkedColor = this.props.titleStyle.checkedColor
+      ? this.props.titleStyle.checkedColor
+      : this.props.titleStyle.color;
+    const uncheckedColor = this.props.titleStyle.uncheckedColor
+      ? this.props.titleStyle.uncheckedColor
+      : this.props.titleStyle.color;
+    // console.log('handleChnage', this.props.titleStyle);
+    if (initialValue === true) {
```
Comment (mahima-vs) [inline examples/src/components/Checkbox.js:44]: if (initialValue) {
// true
} else {
// false
}
=> "initialValue" variable always gives a boolean value so no need to check with equality symbol. Simply use above syntax.

File: examples/src/theme/checkboxTheme.js [hunk 15]
```diff
@@ -0,0 +1,23 @@
+import {getThemeStyle} from '../utility/utils';
+
+// function to change old style without theme to style with theme
+
+export const applyThemeOnCheckboxStyle = (styleWithouthTheme, theme) => {
```
Comment (mahima-vs) [inline examples/src/theme/checkboxTheme.js:5]: I have discuss the code for handling of theme style, pls do the changes.

File: examples/src/index.js [hunk 16]
```diff
@@ -20,38 +20,43 @@ export default class UiBuilder extends React.Component {
...
-              placement="left"
-              leftComponent={{ text: 'Create an account to checkout', style: { color: 'black', fontSize: 20}}}
-              // rightComponent={{ icon: 'cross', color: 'black' }}
-            />
-            {component.childrens.map(componentData => {
-              return this.selectComponent(componentData);
-            })}
-          </View>
-        )
+      // case 'button':
```
Comment (mahima-vs) [inline examples/src/index.js:27]: Why are you commented this piece?

File: examples/src/components/button.js [hunk 17]
```diff
@@ -1,89 +1,96 @@
...
-        // Customize button style
-        const shapeStyle = this.props.showCircle ? { ...shapeStyles.circleShapeView } :
-            this.props.ShowSquare ? { ...shapeStyles.squareShapeView } :
-                this.props.showRectangle ? { ...shapeStyles.rectangleShapeView } :
-                    this.props.showTriangle ? { ...shapeStyles.triangleShapeView } : null;
+//     this.state.props.buttonStyle = shapeStyle
+//       ? {...this.state.props.buttonStyle, ...shapeStyle}
+//       : this.state.props.buttonStyle;
 
+//     // Applying theme on button style
```
Comment (mahima-vs) [inline examples/src/components/button.js:39]: Why are you commented this piece?

File: examples/src/components/Checkbox.js [hunk 18]
```diff
@@ -0,0 +1,121 @@
...
+      });
+    }
+  };
+  render() {
+    let props = this.props;
+    //console.log(this.state);
+    props.titleStyle.color = this.state.color
+      ? this.state.color
+      : props.titleStyle.color;
+    props = theme ? applyThemeOnCheckboxStyle(props, theme) : props;
```
Comment (mahima-vs) [inline examples/src/components/Checkbox.js:63]: props = theme ? applyThemeOnCheckboxStyle(props, theme) : props;
=> 1. why not we are assigning back to `this.props`
=> 2. Add handling of theme piece in componentWillMount method or any other method that will call before render method. 
It's not a good practice to write handling piece in render.

Comment (saksham695) [reply examples/src/components/Checkbox.js:63]: done

File: examples/src/components/button.js (hunk 17 above)
Comment (saksham695) [reply examples/src/components/button.js:39]: actually this component was not working before and i was unable to detect from where it was coming .So one by one i started commenting things first and then detected.

File: examples/src/theme/checkboxTheme.js (hunk 15 above)
Comment (saksham695) [reply examples/src/theme/checkboxTheme.js:5]: changed done

File: examples/src/components/Checkbox.js (hunk 14 above)
Comment (saksham695) [reply examples/src/components/Checkbox.js:44]: done

File: examples/src/components/Checkbox.js (hunk 13 above)
Comment (saksham695) [reply examples/src/components/Checkbox.js:36]: done

File: examples/src/components/Checkbox.js (hunk 12 above)
Comment (saksham695) [reply examples/src/components/Checkbox.js:35]: changed to handleCheckboxStatus

File: examples/src/components/Checkbox.js (hunk 11 above)
Comment (saksham695) [reply examples/src/components/Checkbox.js:43]: code cleanup means removing  unnecessary comments or something i need to change while handling this logic

File: examples/src/components/Checkbox.js (hunk 10 above)
Comment (saksham695) [reply examples/src/components/Checkbox.js:19]: done

File: examples/src/theme/checkboxTheme.js (hunk 8 above)
Comment (saksham695) [reply examples/src/theme/checkboxTheme.js]: done .
These files are deleted from codebase now

File: examples/src/utility/utils.js [hunk 19]
```diff
@@ -1,27 +1,50 @@
...
+
+export const applyTheme = (componentStyle, theme) => {
+  const existingStyle = {...componentStyle};
+  const themeStyle = getThemeStyle(theme);
+  // get changes according to theme in three objects
+  const keys = Object.keys(themeStyle);
+
+  // iterate over each key and checks if style is defined then change the style according to theme
+
+  keys.forEach((key) => {
```
Comment (mahima-vs) [inline examples/src/utility/utils.js:35]: Use Object.keys(themeStyle) directly.

File: examples/src/utility/utils.js [hunk 20]
```diff
@@ -23,3 +23,28 @@ const darkTheme = {
...
+// function will be called when user defined any theme and will change the style of component .
+
+export const applyTheme = (componentStyle, theme) => {
+  const existingStyle = {...componentStyle};
+  const themeStyle = getThemeStyle(theme);
+
+  // iterate over each key and checks if style is defined then change the style according to theme
+
+  Object.keys(themeStyle).forEach((key) => {
+    if (existingStyle.checkboxStyle && existingStyle.checkboxStyle[key]) {
```
Comment (mahima-vs) [inline examples/src/utility/utils.js:36]: After running this line, check the checkboxStyle, this line will suppose to add a key if key is not there in a checkboxStyle, instead our usecase is to update the value of a key if ```key``` is there in existing style.

**Pull request merged**